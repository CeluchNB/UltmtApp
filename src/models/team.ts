import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { LocalUser, Team } from '../types/team'

export const DisplayTeamSchema: Realm.ObjectSchema = {
    name: 'DisplayTeam',
    embedded: true,
    properties: {
        _id: 'string',
        place: 'string',
        name: 'string',
        teamname: 'string',
        seasonStart: 'string',
        seasonEnd: 'string',
    },
}

export const GuestTeamSchema: Realm.ObjectSchema = {
    name: 'GuestTeam',
    embedded: true,
    properties: {
        _id: 'string?',
        place: 'string?',
        name: 'string',
        teamname: 'string?',
        seasonStart: 'string?',
        seasonEnd: 'string?',
    },
}

export class TeamSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Team',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            place: 'string',
            name: 'string',
            teamname: 'string',
            managers: { type: 'list', objectType: 'DisplayUser' },
            players: { type: 'list', objectType: 'DisplayUser' },
            seasonStart: 'string',
            seasonEnd: 'string',
            seasonNumber: 'int',
            continuationId: 'string',
            rosterOpen: 'bool',
        },
    }

    _id: string
    place: string
    name: string
    teamname: string
    managers: DisplayUser[]
    players: LocalUser[]
    seasonStart: string
    seasonEnd: string
    seasonNumber: number
    continuationId: string
    rosterOpen: boolean

    constructor(team: Team) {
        this._id = team._id
        this.place = team.place
        this.name = team.name
        this.teamname = team.teamname
        this.managers = team.managers
        this.players = team.players
        this.seasonStart = team.seasonStart
        this.seasonEnd = team.seasonEnd
        this.seasonNumber = team.seasonNumber
        this.continuationId = team.continuationId
        this.rosterOpen = team.rosterOpen
    }
}
