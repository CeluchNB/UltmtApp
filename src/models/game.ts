import { DisplayUser } from '../types/user'
import { Realm } from '@realm/react'
import { Tournament } from '../types/tournament'
import { CreateGame, Game, GameStatus, PointStats } from '../types/game'
import { DisplayTeam, GuestTeam } from '../types/team'

export class GameSchema {
    static schema: Realm.ObjectSchema = {
        name: 'Game',
        primaryKey: '_id',
        properties: {
            _id: 'string',
            creator: 'DisplayUser',
            teamOne: 'DisplayTeam',
            teamTwo: 'GuestTeam',
            teamTwoDefined: 'bool',
            scoreLimit: 'int',
            halfScore: 'int',
            startTime: 'date',
            softcapMins: 'int',
            hardcapMins: 'int',
            playersPerPoint: 'int',
            timeoutPerHalf: 'int',
            floaterTimeout: 'bool',
            tournament: 'Tournament?',
            teamOneScore: 'int',
            teamTwoScore: 'int',
            teamOnePlayers: 'DisplayUser[]',
            teamTwoPlayers: 'DisplayUser[]',
            resolveCode: 'string',
            statsPoints: 'PointStats[]',
            offline: 'bool',
            teamOneStatus: 'string',
            teamTwoStatus: 'string',
        },
    }

    _id: string
    creator: DisplayUser
    teamOne: DisplayTeam
    teamTwo: GuestTeam
    teamTwoDefined: boolean
    scoreLimit: number
    halfScore: number
    startTime: Date
    softcapMins: number
    hardcapMins: number
    playersPerPoint: number
    timeoutPerHalf: number
    floaterTimeout: boolean
    tournament?: Tournament
    teamOneScore: number
    teamTwoScore: number
    teamOnePlayers: DisplayUser[]
    teamTwoPlayers: DisplayUser[]
    resolveCode: string
    statsPoints: PointStats[]
    offline: boolean
    teamOneStatus: GameStatus
    teamTwoStatus: GameStatus

    constructor(
        game: Game,
        offline: boolean = false,
        statsPoints: PointStats[] = [],
    ) {
        this._id = game._id
        this.creator = game.creator
        this.teamOne = game.teamOne
        this.teamTwo = game.teamTwo
        this.teamTwoDefined = game.teamTwoDefined
        this.scoreLimit = game.scoreLimit
        this.halfScore = game.halfScore
        this.startTime = game.startTime
        this.softcapMins = game.softcapMins
        this.hardcapMins = game.hardcapMins
        this.playersPerPoint = game.playersPerPoint
        this.timeoutPerHalf = game.timeoutPerHalf
        this.floaterTimeout = game.floaterTimeout
        this.tournament = game.tournament
        this.teamOneScore = game.teamOneScore
        this.teamTwoScore = game.teamTwoScore
        this.teamOnePlayers = game.teamOnePlayers
        this.teamTwoPlayers = game.teamTwoPlayers
        this.resolveCode = game.resolveCode || ''
        this.statsPoints = statsPoints
        this.offline = offline
        this.teamOneStatus = game.teamOneStatus
        this.teamTwoStatus = game.teamTwoStatus
    }

    static createOfflineGame(game: CreateGame, teamOnePlayers: DisplayUser[]) {
        return {
            _id: new Realm.BSON.ObjectID().toHexString(),
            creator: game.creator,
            teamOne: game.teamOne,
            teamTwo: game.teamTwo,
            teamTwoDefined: game.teamTwoDefined,
            scoreLimit: game.scoreLimit,
            halfScore: game.halfScore,
            startTime: new Date(game.startTime),
            softcapMins: game.softcapMins,
            hardcapMins: game.hardcapMins,
            playersPerPoint: game.playersPerPoint,
            timeoutPerHalf: game.timeoutPerHalf,
            floaterTimeout: game.floaterTimeout,
            tournament: game.tournament,
            teamOneScore: 0,
            teamTwoScore: 0,
            teamOneActive: true,
            teamTwoActive: false,
            teamOnePlayers: teamOnePlayers,
            teamTwoPlayers: [],
            resolveCode: '',
            statsPoints: [],
            offline: true,
        }
    }
}
export const PointStatsSchema: Realm.ObjectSchema = {
    name: 'PointStats',
    embedded: true,
    properties: {
        _id: 'string',
        pointStats: 'PlayerPointStats[]',
    },
}

export const PlayerPointStatsSchema: Realm.ObjectSchema = {
    name: 'PlayerPointStats',
    embedded: true,
    properties: {
        _id: 'string',
        firstName: 'string',
        lastName: 'string',
        username: 'string?',
        pointsPlayed: 'int',
        goals: 'int',
        assists: 'int',
        turnovers: 'int',
        blocks: 'int',
    },
}
