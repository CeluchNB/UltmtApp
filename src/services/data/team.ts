import * as Constants from '../../utils/constants'
import { ApiError } from '../../types/services'
import { throwApiError } from '../../utils/service-utils'
import { CreateTeam, Team } from '../../types/team'
import {
    createTeam as networkCreateTeam,
    getManagedTeam as networkGetManagedTeam,
    getTeam as networkGetTeam,
    removePlayer as networkRemovePlayer,
    rollover as networkRollover,
    searchTeam as networkSearchTeam,
    toggleRosterStatus as networkToggleRosterStatus,
} from '../network/team'

/**
 * Method to create team
 * @param token token of user creating team
 * @param data create team data
 * @returns created team
 * @throws error if team is not created successfully
 */
export const createTeam = async (
    token: string,
    data: CreateTeam,
): Promise<Team> => {
    try {
        const response = await networkCreateTeam(token, data)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.CREATE_TEAM_ERROR)
    }
}

/**
 * Method to search for teams by place, name, or username
 * @param term search term
 * @returns list of teams matching term
 * @throws error if backend returns an error
 */
export const searchTeam = async (term: string): Promise<Team[]> => {
    try {
        if (term.length < 3) {
            throw new ApiError('Not enough characters to search.')
        }
        const response = await networkSearchTeam(term)
        return response.data
    } catch (error) {
        return throwApiError(error, Constants.SEARCH_ERROR)
    }
}

/**
 * Method to get team that the current user manages
 * @param token jwt of current user
 * @param id team id
 * @returns team object
 * @throws error if backend returns error
 */
export const getManagedTeam = async (
    token: string,
    id: string,
): Promise<Team> => {
    try {
        const response = await networkGetManagedTeam(token, id)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.GET_TEAM_ERROR)
    }
}

/**
 * Method to get a public team
 * @param id id of team to get
 * @returns team object
 * @throws error if backend returns an error
 */
export const getTeam = async (id: string): Promise<Team> => {
    try {
        const response = await networkGetTeam(id)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.GET_TEAM_ERROR)
    }
}

/**
 * Method to change a team's roster status
 * @param token jwt of user changing team
 * @param id id of team to update
 * @param open boolean for team's roster status
 * @returns updated team object
 * @throws error if backend returns error
 */
export const toggleRosterStatus = async (
    token: string,
    id: string,
    open: boolean,
): Promise<Team> => {
    try {
        const response = await networkToggleRosterStatus(token, id, open)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.EDIT_TEAM_ERROR)
    }
}

/**
 * Method to remove a player from a team
 * @param token token of current user
 * @param teamId team player is on
 * @param userId id of player to remove
 * @returns updated team object
 * @throws error if backend returns an error
 */
export const removePlayer = async (
    token: string,
    teamId: string,
    userId: string,
): Promise<Team> => {
    try {
        const response = await networkRemovePlayer(token, teamId, userId)
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.EDIT_TEAM_ERROR)
    }
}

/**
 * Method to start a new season for a team. After this method is called,
 * the old team is no longer editable.
 * @param token jwt of manager of team
 * @param teamId id of team to move over
 * @param copyPlayers boolean for deciding if current roster should be kept
 * @param seasonStart start year for season
 * @param seasonEnd start end for season
 * @returns newly created team object
 * @throws error if backend returns an error
 */
export const rollover = async (
    token: string,
    teamId: string,
    copyPlayers: boolean,
    seasonStart: string,
    seasonEnd: string,
): Promise<Team> => {
    try {
        const response = await networkRollover(
            token,
            teamId,
            copyPlayers,
            seasonStart,
            seasonEnd,
        )
        const { team } = response.data
        return team
    } catch (error) {
        return throwApiError(error, Constants.EDIT_TEAM_ERROR)
    }
}
