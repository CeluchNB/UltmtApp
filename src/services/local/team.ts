import { Realm } from '@realm/react'
import { Team } from '../../types/team'
import { TeamSchema } from '../../models'
import { getRealm } from '../../models/realm'

export const saveTeams = async (teams: Team[]) => {
    const realm = await getRealm()
    realm.write(() => {
        for (const team of teams) {
            realm.create('Team', team, Realm.UpdateMode.Modified)
        }
    })
}

export const getManagingTeams = async (id: string): Promise<Team[]> => {
    const realm = await getRealm()
    const teams = await realm.objects<TeamSchema>('Team')
    return teams
        .filter(t => t.managers.map(m => m._id).includes(id))
        .map(t => parseTeam(t))
}

const parseTeam = (schema: TeamSchema): Team => {
    return JSON.parse(
        JSON.stringify({
            _id: schema._id,
            place: schema.place,
            name: schema.name,
            teamname: schema.teamname,
            managers: schema.managers,
            players: schema.players,
            seasonStart: schema.seasonStart,
            seasonEnd: schema.seasonEnd,
            seasonNumber: schema.seasonNumber,
            continuationId: schema.continuationId,
            rosterOpen: schema.rosterOpen,
            requests: [],
            games: [],
        }),
    )
}
