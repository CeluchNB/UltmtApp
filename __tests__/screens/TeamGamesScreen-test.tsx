import * as GameData from '../../src/services/data/game'
import { DisplayTeam } from '../../src/types/team'
import { Game } from '../../src/types/game'
import { Provider } from 'react-redux'
import React from 'react'
import { TeamGameProps } from '../../src/types/navigation'
import TeamGameScreen from '../../src/screens/TeamGamesScreen'
import { fetchProfileData } from '../../fixtures/data'
import { setProfile } from '../../src/store/reducers/features/account/accountReducer'
import store from '../../src/store/store'
import { fireEvent, render, waitFor } from '@testing-library/react-native'

export const getGame = (team: DisplayTeam): Game => {
    return {
        _id: 'game1',
        creator: {
            _id: 'user1',
            firstName: 'First 1',
            lastName: 'Last 1',
            username: 'first1last1',
        },
        teamOne: team,
        teamTwo: { name: 'Sockeye' },
        teamTwoDefined: false,
        scoreLimit: 15,
        halfScore: 8,
        teamOneScore: 3,
        teamTwoScore: 0,
        startTime: new Date('2022-10-12'),
        softcapMins: 75,
        hardcapMins: 90,
        teamOneActive: true,
        teamTwoActive: false,
        playersPerPoint: 7,
        resolveCode: '111111',
        timeoutPerHalf: 1,
        floaterTimeout: true,
        teamOnePlayers: [],
        teamTwoPlayers: [],
        tournament: undefined,
        points: [],
    }
}

const props: TeamGameProps = {
    navigation: {
        navigate: jest.fn(),
    } as any,
    route: {} as any,
}

describe('TeamGamesScreen', () => {
    beforeEach(() => {
        jest.spyOn(GameData, 'getGamesByTeam').mockImplementation(
            async (teamId: string) => {
                if (teamId === fetchProfileData.managerTeams[0]._id) {
                    return [getGame(fetchProfileData.managerTeams[0])]
                }
                return [getGame(fetchProfileData.playerTeams[0])]
            },
        )
        store.dispatch(setProfile(fetchProfileData))
    })
    it('matches snapshot', async () => {
        const snapshot = render(
            <Provider store={store}>
                <TeamGameScreen {...props} />
            </Provider>,
        )
        await waitFor(async () => {
            expect(snapshot.getAllByText(`3 - 0`).length).toBe(2)
        })
        expect(snapshot.toJSON()).toMatchSnapshot()
    })

    it('navigates on press', async () => {
        const { getAllByText } = render(
            <Provider store={store}>
                <TeamGameScreen {...props} />
            </Provider>,
        )
        await waitFor(async () => {
            expect(getAllByText(`3 - 0`).length).toBe(2)
        })
        const team = getAllByText('3 - 0')[0]
        fireEvent.press(team)

        expect(props.navigation.navigate).toHaveBeenCalledWith('Tabs', {
            screen: 'Games',
            params: {
                screen: 'ViewGame',
                params: { gameId: 'game1' },
            },
        })
    })
})
