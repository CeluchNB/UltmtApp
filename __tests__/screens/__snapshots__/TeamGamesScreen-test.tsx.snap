// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamGamesScreen matches snapshot 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#303030",
      "height": "100%",
    }
  }
>
  <View
    style={
      {
        "alignSelf": "center",
        "width": "80%",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#ffd600",
            "fontSize": 61.64383561643835,
            "fontWeight": "bold",
          },
          undefined,
        ]
      }
    >
      Your Games
    </Text>
    <RCTScrollView
      data={
        [
          {
            "data": [
              {
                "_id": "game1",
                "creator": {
                  "_id": "user1",
                  "firstName": "First 1",
                  "lastName": "Last 1",
                  "username": "first1last1",
                },
                "floaterTimeout": true,
                "halfScore": 8,
                "hardcapMins": 90,
                "playersPerPoint": 7,
                "points": [],
                "resolveCode": "111111",
                "scoreLimit": 15,
                "softcapMins": 75,
                "startTime": 2022-10-12T00:00:00.000Z,
                "teamOne": {
                  "_id": "id4",
                  "name": "Name4",
                  "place": "Place4",
                  "seasonEnd": "2022",
                  "seasonStart": "2022",
                  "teamname": "place4name4",
                },
                "teamOneActive": true,
                "teamOnePlayers": [],
                "teamOneScore": 3,
                "teamTwo": {
                  "name": "Sockeye",
                },
                "teamTwoActive": false,
                "teamTwoDefined": false,
                "teamTwoPlayers": [],
                "teamTwoScore": 0,
                "timeoutPerHalf": 1,
                "tournament": undefined,
              },
            ],
            "index": 0,
            "title": "Place4 Name4",
          },
          {
            "data": [
              {
                "_id": "game1",
                "creator": {
                  "_id": "user1",
                  "firstName": "First 1",
                  "lastName": "Last 1",
                  "username": "first1last1",
                },
                "floaterTimeout": true,
                "halfScore": 8,
                "hardcapMins": 90,
                "playersPerPoint": 7,
                "points": [],
                "resolveCode": "111111",
                "scoreLimit": 15,
                "softcapMins": 75,
                "startTime": 2022-10-12T00:00:00.000Z,
                "teamOne": {
                  "_id": "id1",
                  "name": "Name1",
                  "place": "Place1",
                  "seasonEnd": "2019",
                  "seasonStart": "2019",
                  "teamname": "place1name1",
                },
                "teamOneActive": true,
                "teamOnePlayers": [],
                "teamOneScore": 3,
                "teamTwo": {
                  "name": "Sockeye",
                },
                "teamTwoActive": false,
                "teamTwoDefined": false,
                "teamTwoPlayers": [],
                "teamTwoScore": 0,
                "timeoutPerHalf": 1,
                "tournament": undefined,
              },
            ],
            "index": 1,
            "title": "Place1 Name1",
          },
          {
            "data": [
              {
                "_id": "game1",
                "creator": {
                  "_id": "user1",
                  "firstName": "First 1",
                  "lastName": "Last 1",
                  "username": "first1last1",
                },
                "floaterTimeout": true,
                "halfScore": 8,
                "hardcapMins": 90,
                "playersPerPoint": 7,
                "points": [],
                "resolveCode": "111111",
                "scoreLimit": 15,
                "softcapMins": 75,
                "startTime": 2022-10-12T00:00:00.000Z,
                "teamOne": {
                  "_id": "id1",
                  "name": "Name1",
                  "place": "Place1",
                  "seasonEnd": "2019",
                  "seasonStart": "2019",
                  "teamname": "place1name1",
                },
                "teamOneActive": true,
                "teamOnePlayers": [],
                "teamOneScore": 3,
                "teamTwo": {
                  "name": "Sockeye",
                },
                "teamTwoActive": false,
                "teamTwoDefined": false,
                "teamTwoPlayers": [],
                "teamTwoScore": 0,
                "timeoutPerHalf": 1,
                "tournament": undefined,
              },
            ],
            "index": 2,
            "title": "Place2 Name2",
          },
          {
            "data": [
              {
                "_id": "game1",
                "creator": {
                  "_id": "user1",
                  "firstName": "First 1",
                  "lastName": "Last 1",
                  "username": "first1last1",
                },
                "floaterTimeout": true,
                "halfScore": 8,
                "hardcapMins": 90,
                "playersPerPoint": 7,
                "points": [],
                "resolveCode": "111111",
                "scoreLimit": 15,
                "softcapMins": 75,
                "startTime": 2022-10-12T00:00:00.000Z,
                "teamOne": {
                  "_id": "id1",
                  "name": "Name1",
                  "place": "Place1",
                  "seasonEnd": "2019",
                  "seasonStart": "2019",
                  "teamname": "place1name1",
                },
                "teamOneActive": true,
                "teamOnePlayers": [],
                "teamOneScore": 3,
                "teamTwo": {
                  "name": "Sockeye",
                },
                "teamTwoActive": false,
                "teamTwoDefined": false,
                "teamTwoPlayers": [],
                "teamTwoScore": 0,
                "timeoutPerHalf": 1,
                "tournament": undefined,
              },
            ],
            "index": 3,
            "title": "Place3 Name3",
          },
        ]
      }
      getItem={[Function]}
      getItemCount={[Function]}
      keyExtractor={[Function]}
      onContentSizeChange={[Function]}
      onLayout={[Function]}
      onMomentumScrollBegin={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={
        [
          0,
          3,
          6,
          9,
        ]
      }
    >
      <View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              {
                "color": "#ffd600",
                "fontSize": 61.64383561643835,
                "fontWeight": "bold",
              }
            }
          >
            Place4 Name4
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#ababab",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  vs. Sockeye
                </Text>
                <Text
                  style={
                    {
                      "color": "#ffd600",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  3 - 0
                </Text>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "height": 40,
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="go-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#ffd600",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰅂
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              {
                "color": "#ffd600",
                "fontSize": 61.64383561643835,
                "fontWeight": "bold",
              }
            }
          >
            Place1 Name1
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#ababab",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  vs. Sockeye
                </Text>
                <Text
                  style={
                    {
                      "color": "#ffd600",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  3 - 0
                </Text>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "height": 40,
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="go-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#ffd600",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰅂
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              {
                "color": "#ffd600",
                "fontSize": 61.64383561643835,
                "fontWeight": "bold",
              }
            }
          >
            Place2 Name2
          </Text>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        >
          <View
            style={
              {
                "display": "flex",
                "flexDirection": "row",
                "marginBottom": 5,
              }
            }
          >
            <View
              style={
                {
                  "alignSelf": "center",
                  "flex": 1,
                }
              }
            >
              <View
                accessible={true}
                collapsable={false}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  {
                    "opacity": 1,
                  }
                }
              >
                <Text
                  style={
                    {
                      "color": "#ababab",
                      "fontSize": 41.0958904109589,
                      "fontWeight": "bold",
                    }
                  }
                >
                  vs. Sockeye
                </Text>
                <Text
                  style={
                    {
                      "color": "#ffd600",
                      "fontSize": 20.54794520547945,
                      "fontWeight": "bold",
                    }
                  }
                >
                  0 - 3
                </Text>
              </View>
            </View>
            <View
              collapsable={false}
              style={
                {
                  "alignSelf": undefined,
                  "bottom": undefined,
                  "left": undefined,
                  "position": undefined,
                  "right": undefined,
                  "shadowColor": "#000",
                  "shadowOffset": {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                  "top": undefined,
                }
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "shadowColor": "#000",
                    "shadowOffset": {
                      "height": 0,
                      "width": 0,
                    },
                    "shadowOpacity": 0,
                    "shadowRadius": 0,
                  }
                }
              >
                <View
                  collapsable={false}
                  style={
                    {
                      "alignItems": "center",
                      "backgroundColor": undefined,
                      "borderColor": "rgba(121, 116, 126, 1)",
                      "borderRadius": 20,
                      "borderWidth": 0,
                      "elevation": 0,
                      "height": 40,
                      "justifyContent": "center",
                      "margin": 6,
                      "overflow": "hidden",
                      "width": 40,
                    }
                  }
                >
                  <View
                    accessibilityComponentType="button"
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "disabled": false,
                      }
                    }
                    accessibilityTraits="button"
                    accessible={true}
                    centered={true}
                    collapsable={false}
                    focusable={true}
                    hitSlop={
                      {
                        "bottom": 6,
                        "left": 6,
                        "right": 6,
                        "top": 6,
                      }
                    }
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        false,
                        [
                          {
                            "alignItems": "center",
                            "flexGrow": 1,
                            "justifyContent": "center",
                          },
                          {
                            "borderRadius": 20,
                          },
                        ],
                      ]
                    }
                    testID="go-button"
                  >
                    <Text
                      accessibilityElementsHidden={true}
                      allowFontScaling={false}
                      importantForAccessibility="no-hide-descendants"
                      pointerEvents="none"
                      selectable={false}
                      style={
                        [
                          {
                            "color": "#ffd600",
                            "fontSize": 24,
                          },
                          [
                            {
                              "lineHeight": 24,
                              "transform": [
                                {
                                  "scaleX": 1,
                                },
                              ],
                            },
                            {
                              "backgroundColor": "transparent",
                            },
                          ],
                          {
                            "fontFamily": "Material Design Icons",
                            "fontStyle": "normal",
                            "fontWeight": "normal",
                          },
                          {},
                        ]
                      }
                    >
                      󰅂
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={null}
        />
        <View
          onLayout={[Function]}
          style={null}
        >
          <Text
            style={
              {
                "color": "#ffd600",
                "fontSize": 61.64383561643835,
                "fontWeight": "bold",
              }
            }
          >
            Place3 Name3
          </Text>
        </View>
        <View
          style={
            {
              "height": 0,
            }
          }
        />
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
